### PR TITLE
remove lifetimes that can be elided (clippy::needless_lifetimes)

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

**Comment by eddyb** [comment]
cc @rust-lang/compiler Don't we have a regular lint that also does this?

I remember wanting to do something like this then hitting some sort of issue.

File: src/librustc_codegen_llvm/common.rs [hunk 1]
```diff
@@ -320,7 +320,7 @@ impl ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
     }
 }
 
-pub fn val_ty(v: &'ll Value) -> &'ll Type {
+pub fn val_ty(v: &Value) -> &'_ Type {
```
Comment (eddyb) [inline src/librustc_codegen_llvm/common.rs]: This is weird, shouldn't be `-> &Type`?

File: src/librustc_codegen_llvm/llvm/mod.rs [hunk 2]
```diff
@@ -203,7 +203,7 @@ pub fn get_param(llfn: &'a Value, index: c_uint) -> &'a Value {
 }
 
 /// Safe wrapper for `LLVMGetValueName2` into a byte slice
-pub fn get_value_name(value: &'a Value) -> &'a [u8] {
+pub fn get_value_name(value: &Value) -> &'_ [u8] {
```
Comment (eddyb) [inline src/librustc_codegen_llvm/llvm/mod.rs]: More examples here, maybe search for `-> &'_ ` and replace them?

File: src/librustc_driver/pretty.rs [hunk 3]
```diff
@@ -315,11 +315,11 @@ impl<'b, 'tcx> HirPrinterSupport<'tcx> for TypedAnnotation<'b, 'tcx> {
         &self.tcx.sess
     }
 
-    fn hir_map<'a>(&'a self) -> Option<&'a hir_map::Map<'tcx>> {
+    fn hir_map(&self) -> Option<&hir_map::Map<'tcx>> {
         Some(&self.tcx.hir())
     }
 
-    fn pp_ann<'a>(&'a self) -> &'a dyn pprust_hir::PpAnn {
+    fn pp_ann(&self) -> &dyn pprust_hir::PpAnn {
```
Comment (eddyb) [inline src/librustc_driver/pretty.rs]: Hmm none of the changes in this file seem to create `&'_ `.

**Review by eddyb — APPROVED** [review]

**Comment by eddyb** [comment]
Ah, I must be thinking of `single_use_lifetimes`, see https://github.com/rust-lang/rust/issues/53738#issuecomment-501858176.

(coincidentally got notified of that issue just now)

**Comment by matthiaskrgr** [comment]
Right, removed the redundant anonymous lifetimes

**Review by eddyb — APPROVED** [review]

**Pull request merged**